### ISSUE TITLE
How do I get the cluster number to which each sample belongs? 

### ISSUE THREAD

**Issue by sumanth-bmsce** [JustGlowing/minisom#64]
Hi,

Is there any way I can get the cluster number to which each datapoint in that dataset belongs to?

**Comment by JustGlowing** [comment]
Hi, since this is a recurring question I decided to add an example here: https://github.com/JustGlowing/minisom/blob/master/examples/Clustering.ipynb

**Comment by sumanth-bmsce** [comment]
Hi,

Thanks for quickly answering my question. Depending on the dataset info i.e. no of clusters and number of data points in each class, we can calculate the centroids for the coordinates returned by som.winner and use it to find the cluster number and calculate the clustering accuracy?

**Comment by JustGlowing** [comment]
Of course you can. The centroids are the weights of the som  (you can also
compute the centroids by averaging the data in each cluster).

You can use any metric for computing the accuracy of your clustering
method, minisom offers the quantization error.

**Comment by sumanth-bmsce** [comment]
Is it possible that I can contribute computing the cluster numbers and clustering accuracy in the way which I described before to this repo?

**Comment by JustGlowing** [comment]
Of course, you have to open a pull request. Google is your friends for that
;)

**Comment by JustGlowing** [comment]
PS: I updated the example so that also the centroids are plotted.

**Comment by Edmond-Lee-Zse-Wong** [comment]
Hi,
my original dataset has no labels, is there any way to sign every datapoints after som training?
PS: I use a 9*9 som and the distance map shows that the dataset is approximately divided to 2 clusters
Thx !

**Comment by JustGlowing** [comment]
hi @Edmond-Lee-Zse-Wong check the example linked above.

**Comment by Edmond-Lee-Zse-Wong** [comment]
Hi
I've checked it last night but the problem hasn't been tackled. As you can see 11 winner neurons are activated, but my dataset has no possibility to be divided to 11 clusters since it is a simple bidimensional dataset(236 points). The som size is calculated by an experiance formulation: size = 5√N, where N is the number of datapoints. The only solution is to reduce the size of som?
![Figure_1](https://user-images.githubusercontent.com/56387202/112631486-e76c1180-8e71-11eb-99b6-07f59ececa76.png)
![Figure_2](https://user-images.githubusercontent.com/56387202/112631526-f652c400-8e71-11eb-9132-78c4754e1e37.png)

**Comment by JustGlowing** [comment]
It's likely that some clusters are spread across different neurons. The som size is not adequate for the task that you are trying to perform.

**Comment by Edmond-Lee-Zse-Wong** [comment]
Yes I think so, maybe I should reduce the size to 2 or 3. 
Thx for your help !!!
